crates/router/src/core/admin.rs

```diff
@@ -269,18 +281,132 @@ pub async fn create_merchant_account(
+                .change_context(errors::ApiErrorResponse::InvalidDataValue {
+                    field_name: "metadata",
+                })
+        })
+        .transpose()?
+        .map(Secret::new);
+
+    let db = state.store.as_ref();
+
+    db.find_organization_by_org_id(&req.organization_id)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::GenericNotFoundError {
+            message: "organization with the given id does not exist".to_string(),
+        })?;
+
+    let master_key = db.get_master_key();
+
+    let key = services::generate_aes256_key()
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Unable to generate aes 256 key")?;
+
+    let key_store = domain::MerchantKeyStore {
+        merchant_id: req.id.clone(),
+        key: domain_types::encrypt(key.to_vec().into(), master_key)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failed to decrypt data from key store")?,
+        created_at: date_time::now(),
+    };
+
+    db.insert_merchant_key_store(key_store.clone(), &master_key.to_vec().into())
+        .await
+        .to_duplicate_response(errors::ApiErrorResponse::DuplicateMerchantAccount)?;
+
+    let primary_business_details = Vec::<api_models::admin::PrimaryBusinessDetails>::new()
+        .encode_to_value()
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+    let merchant_account = async {
+        Ok::<_, error_stack::Report<common_utils::errors::CryptoError>>(domain::MerchantAccount {
+            merchant_id: req.id,
+            merchant_name: req
+                .merchant_name
+                .async_lift(|inner| domain_types::encrypt_optional(inner, &key))
+                .await?,
+            merchant_details: merchant_details
+                .async_lift(|inner| domain_types::encrypt_optional(inner, &key))
+                .await?,
+            return_url: None,
+            webhook_details: None,
+            routing_algorithm: Some(serde_json::json!({
+                "algorithm_id": null,
+                "timestamp": 0
+            })),
+            sub_merchants_enabled: None,
+            parent_merchant_id: None,
+            enable_payment_response_hash: true,
+            payment_response_hash_key: None,
+            redirect_to_merchant_with_http_post: true,
+            publishable_key,
+            locker_id: None,
+            metadata,
+            storage_scheme: MerchantStorageScheme::PostgresOnly,
+            primary_business_details,
+            created_at: date_time::now(),
+            modified_at: date_time::now(),
+            intent_fulfillment_time: None,
+            frm_routing_algorithm: None,
+            #[cfg(feature = "payouts")]
+            payout_routing_algorithm: None,
+            #[cfg(not(feature = "payouts"))]
+            payout_routing_algorithm: None,
+            id: None,
```

Comment:
We should populate this field too.

Reply:
The type for this field currently is Int, the type should be changed and then we can populate uuid in this field